enhancements/1845-nfd-image-compatibility/README.md

```diff
@@ -0,0 +1,184 @@
+Cloud-native technologies are being adopted by high-demand industries where container compatibility is critical for service performance and cluster preparation.
+This proposal introduces the concept of NFD image compatibility metadata.
+NFD features via NodeFeatureRules CRs can be effectively added to images to specify requirements for a host or operating system.
+
+The document has been prepared based on the experience and progress of the [OCI Image Compatibility working group](https://github.com/opencontainers/wg-image-compatibility/tree/main/docs/proposals).
+
+## Motivation
+
+Image compatibility metadata will help container image authors describe compatibility requirements in a standardized way.
+This metadata will be uploaded with the image to the image registry.
+As a result, container compatibility requirements will become discoverable and programmable, supporting various consumers and use cases where applications require a specific compatible environment.
+
+### Goals
+
+#### Phase 1
+
+- Use existing NFD features via the NodeFeatureRule API to describe container image requirements.
+- Create a new OCI artifact type for compatibility metadata.
+- Allow verification of node compatibilitym including nodes that are not yet part of the k8s cluster.
+- Add or extend the sources with missing features.
+
+#### Phase 2
+
+Phase 2 involves future prediction and shows the general direction.
+After the completion of Phase 1, either this document should be updated, or a new proposal should be created that considers the following points:
+
+- Update or generate pods with appropriate node selectors via a mutation webhook or a scheduler plugin.
+
+### Non-Goals
+
+- Make image compatibility a hard requirement for the NFD installation/usage.
+- Cover applications ABI compatibility.
+
+## Proposal
+
+Build a new NFD client tool with the following initial scope:
+
+- CRUD OCI artifact.
+- Validate nodes based on provided metadata.
+- Run directly on a host which is not part of the Kubernetes cluster, or run as a Kubernetes job on a Kubernetes node.
+
+### Design Details
+
+#### OCI Artifact
+
+[An OCI artifact](https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidelines-for-artifact-usage) should be created to store image compatibility metadata on the image side.  
+The artifact can be connected with an image over [the subject field](https://github.com/opencontainers/distribution-spec/blob/11b8e3fba7d2d7329513d0cff53058243c334858/spec.md#pushing-manifests-with-subject).
+
+##### Manifest
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.k8s.nfd.image-compatibility.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.k8s.nfd.image-compatibility.spec.v1+json",
```

Comment:
it could be "application/vnd.k8s.nfd.image-compatibility.spec.v1+json" or "application/vnd.k8s.nfd.image-compatibility.spec.v1+yaml" ?  in kubernetes, usually yaml is used.